### ISSUE TITLE
BTDropInResult:fetchDropInResultForAuthorization method for last used card (default payment) returns last ADDED card, not last selected

### ISSUE THREAD

**Issue by andriy-appuchino** [braintree/braintree-ios-drop-in#180]
### General information

* SDK/Library version: 6.5.0 and 7.4.0
* Environment: Sandbox and Production
* iOS Version and Device: iOS 12.4.1 on an iPhone X
* Integration type and version: CocoaPods 1.7.5

### Issue description

`BTDropInResult:fetchDropInResultForAuthorization` method for last used card (default payment) returns last ADDED card, not last selected. It seems there is a bug.
https://developers.braintreepayments.com/guides/drop-in/setup-and-integration/ios/v4#displaying-the-last-used-payment-method

Steps:
1. Add card # 1 using DropIn
- fetchDropInResultForAuthorization returns card # 1 
2. Add card # 2 using DropIn
- fetchDropInResultForAuthorization returns card # 2 
3. **Select card # 1 using DropIn**
- BTDropInControllerHandler returns card # 1 in BTDropInResult
- **fetchDropInResultForAuthorization returns card # 2 :(**

With Android SDK the same problem too.

**Comment by scannillo** [comment]
Hello. Thank you for opening this issue. I tested this and can confirm that `BTDropInResult:fetchDropInResultForAuthorization`'s `result` is indeed the last added card and NOT the last selected card.

I will make a ticket with our team to update our docs to more appropriately reflect this functionality.

**Comment by andriy-appuchino** [comment]
@scannillo But how do we get the last selected card? This is exactly the one that DropIn shows first on the list.

**Comment by scannillo** [comment]
Hello again. I spoke with @crookedneighbor on our team to clarify a few things.

- So the first payment method that you see in the list of vaulted payment methods is not the last selected, but instead is the "default" payment method listed in your customer's vault. If you try adding a few more vaulted payment methods to your testing, you will be able to see how this works (I think the order of your current payment methods was just coincidental). Here is code to see how we fetch the payment methods and specify the first will be the default [here](https://github.com/braintree/braintree_ios/blob/d759f678e045ee4055cbc18d555df637ac1aefc2/BraintreeCore/BTAPIClient.m#L218-L220).

- When using the drop-in to vault, the first added payment method will become the default. 

- If you want to update/change the default payment method, you can do so using the server SDK. Here is a link to our [python SDK example](https://developers.braintreepayments.com/reference/request/payment-method/create/python#new-default-payment-method), but you can utilize this in whichever server SDK you use.

**Comment by andriy-appuchino** [comment]
@scannillo Hello. Thank you for the answer, but I can not agree that the order is coincidental. DropIn itself has the title "Select Payment Method" and a list for selection, and if you select one, then DropIn hides. I tried several dozens of times already, and the first card (payment method) in the list is always the one I last selected.
So, this is the direct purpose of this DropIn not only to add payment method, but also to select the preferred one from the list. Do I misunderstand the purpose of DropIn with its title?

**Comment by scannillo** [comment]
Hello. I have attached a video of our demo app using drop-in to demonstrate that the behavior you see is something I am not able to replicate. (The "Change Payment Method" button is synonymous to "Show Drop-In"). You will see that the first card that I have added, a VISA, card is always shown first. Selecting a card and then closing the drop in does _not_ place it in the front of the vaulted list.

There must be something else going on with your integration to be setting your last used card as a default. I can't really help you more without more context.

![GIF](https://user-images.githubusercontent.com/35243507/65055044-2859a080-d934-11e9-85f9-16d43337accb.gif)

Since the above is expected behavior on our end, to get further help on troubleshooting/integrating with our SDKs, reach out to our support team at https://help.braintreepayments.com. 

Your original issue/suggestion regarding `BTDropInResult:fetchDropInResultForAuthorization` has been added to our backlog!
